contracts/SimpleSwap.sol

```diff
@@ -62,50 +63,63 @@ contract SimpleSwap {
   /// @param beneficiary the beneficiary of the cheque
   /// @param serial the serial number of the cheque
   /// @param amount the (cumulative) amount of the cheque
+  /// @param timeout the check can be cashed timeout seconds in the future
   function _submitChequeInternal(address beneficiary, uint serial, uint amount, uint timeout) internal {
+    ChequeInfo storage cheque = cheques[beneficiary];
     /* ensure serial is increasing */
-    ChequeInfo storage info = cheques[beneficiary];
-    require(serial > info.serial);
-
+    require(serial > cheque.serial,
+    "SimpleSwap: invalid serial");
     /* update the stored info */
-    info.serial = serial;
-    info.amount = amount;
-    /* the check can be cashed timeout seconds in the future */
-    info.timeout = now + timeout;
-
+    cheque.serial = serial;
+    cheque.amount = amount;
+    cheque.timeout = now + timeout;
     /* the channel participants should watch to this event to find out if an older cheque is being submitted */
     emit ChequeSubmitted(beneficiary, serial, amount);
   }
 
-  /// @notice submit a cheque
+  /// @notice submit a cheque by the owner
   /// @param beneficiary the beneficiary of the cheque
   /// @param serial the serial number of the cheque
   /// @param amount the (cumulative) amount of the cheque
-  /// @param sig signature of the owner
-  function submitCheque(address beneficiary, uint serial, uint amount, uint timeout, bytes memory sig) public {
-    /* only allow beneficiary to submit this, otherwise the owner could block cash out by regulary sending 1 wei cheques and resetting the timeout */
-    /* unfortunately this breaks watchtowers, so the timeout mechanism should be changed */
-    require(msg.sender == beneficiary);
-    /* verify signature of the owner */
-    require(owner ==  recover(chequeHash(address(this), beneficiary, serial, amount, timeout), sig));
-    /*  amount needs to be larger. since this can only be called by the beneficiary this is probably not necessary */
-    require(amount > cheques[beneficiary].amount);
+  /// @param timeout the check can be cashed timeout seconds in the future
+  /// @param beneficiarySig signature of the owner
+  function submitChequeOwner(address beneficiary, uint serial, uint amount, uint timeout, bytes memory beneficiarySig) public {
+    require(msg.sender == owner);
```

Comment:
Always add a revert reason (e.g. "not owner") here, otherwise this becomes difficult to debug.

Reply:
Done. Also added comments in all other require statements. Is it an idea to make an issue to make a standard for revert reasons in the SW3 suite?